tools/wifi_scanner.py

```diff
@@ -0,0 +1,177 @@
+            return []
+
+        self.serial.reset_input_buffer()
+        
+        command = '{"commands":[{"command":"scan_networks","data":{}}]}\n'
+        self.serial.write(command.encode())
+        
+        timeout_start = time.time()
+        response_buffer = ""
+        
+        while time.time() - timeout_start < timeout_seconds:
+            if self.serial.in_waiting:
+                data = self.serial.read(self.serial.in_waiting).decode('utf-8', errors='ignore')
+                response_buffer += data
+                
+                # Look for WiFi networks JSON directly (new format)
+                # The scan command now outputs JSON directly followed by command result
+                if '{"networks":[' in response_buffer:
+                    import re
+                    
+                    # Look for the networks JSON pattern that appears first
+                    networks_pattern = r'\{"networks":\[.*?\]\}'
+                    matches = re.findall(networks_pattern, response_buffer, re.DOTALL)
+                    
+                    for match in matches:
+                        try:
+                            # Parse the networks JSON directly
+                            networks_data = json.loads(match)
+                            if "networks" in networks_data:
+                                return networks_data["networks"]
+                        except json.JSONDecodeError:
+                            continue
+                
+                # Also check if we have the command result indicating completion
+                if '{"results":' in response_buffer and '"Networks scanned"' in response_buffer:
+                    # We've received the completion message, parse any networks found
+                    import re
+                    networks_pattern = r'\{"networks":\[.*?\]\}'
+                    matches = re.findall(networks_pattern, response_buffer, re.DOTALL)
+                    
+                    for match in matches:
+                        try:
+                            networks_data = json.loads(match)
+                            if "networks" in networks_data:
+                                return networks_data["networks"]
+                        except json.JSONDecodeError:
+                            continue
+                    
+                    # If we get here, scan completed but no networks found
+                    return []
+            else:
+                time.sleep(0.1)
+        
+        print("Failed to receive clean JSON response. Raw data:")
+        print("=" * 50)
+        print(response_buffer)
+        print("=" * 50)
+        return []
+
+    def close(self):
+        if self.serial:
+            self.serial.close()
+
+def main():
+    import sys
+    import argparse
+    
+    parser = argparse.ArgumentParser(description='ESP32 WiFi Scanner')
+    parser.add_argument('port', nargs='?', default='COM15', help='Serial port (default: COM9)')
```

Comment:
default's COM15 though, then again, I don't think we can rely on a default here, that's one of the arguments that most likely should be a required one